### DISCUSSION TITLE
Consolidate all text elements to a single component in skinning

### DISCUSSION THREAD

**Discussion by LeNitrous** [ppy/osu#25500] [Ideas]
As an alternative to https://github.com/ppy/osu/discussions/25457, it would be great if all text-related components in the skin editor use a single "Label" component where it can be formatted freely. This is similar to the existing `BeatmapAttributeText` component where the label can have a template but consolidates the following elements that serve a similar purpose:
- Accuracy Counter
- Combo Counter
- Score Counter
- Song Progress (progress text)
- BPM counter
- Clicks per second Counter
- Judgement Counter
- Unstable Rate Counter
- Performance Points Counter
- Player Name

This allows more control over labels present in existing default and soon to come user-made skins as well as reducing the number of components that are visually similar (as they only contain only text at most). This only covers text formatting for now but as an aside, it would be even better if it allows custom fonts and markdown-based text formatting. However, that is outside the scope of this discussion.

**Comment by peppy** [comment]
More formatting will come later. They already share a base type which will allow formatting for all text. Combining components is not necessary.
